### PR TITLE
build: create make rpm and make srpm

### PR DESCRIPTION
**Pull request by dougsland**
Make life easier to build packages for devs.
Previously, I had to dig the source code to find
the script.

GH-Issue: https://github.com/containers/hirte/issues/284

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

**Review by engelmi — COMMENTED** [review]
Our primary build system is [meson](mesonbuild.com/) and most targets in the Makefile just delegate to it - so we should keep this file rather slim. However, you are right that the info on how to build the rpm is rather hidden.  

I'd propose to create custom targets to meson, similar to `clang-tidy`:
https://github.com/containers/hirte/blob/main/meson.build#L72C23-L78
There you can also check if `rpmbuild` is installed. Then, change the make targets here to:
```Makefile
rpm: 
   ninja -C {BUILDDIR} build-rpm
```

**Comment by mwperina** [comment]
> Our primary build system is [meson](mesonbuild.com/) and most targets in the Makefile just delegate to it - so we should keep this file rather slim. However, you are right that the info on how to build the rpm is rather hidden.
> 
> I'd propose to create custom targets to meson, similar to `clang-tidy`: https://github.com/containers/hirte/blob/main/meson.build#L72C23-L78 There you can also check if `rpmbuild` is installed. Then, change the make targets here to:
> 
> ```makefile
> rpm: 
>    ninja -C {BUILDDIR} build-rpm
> ```

I think current patch is OK, because meson is executed as a part of build scripts. I don't see a reason why we should call those scripts from meson itself, it would be a bit chicken egg problem

**Review by mwperina — APPROVED** [review]
+1

**Comment by dougsland** [comment]
@engelmi @mwperina Thanks for the comments. I was thinking like: "makefile triggers the real build system (whatever it is)" which in this case is `meson` and generates the `srpm` and `rpm` files. Is that correct?

**Comment by mwperina** [comment]
> @engelmi @mwperina Thanks for the comments. I was thinking like: "makefile triggers the real build system (whatever it is)" which in this case is `meson` and generates the `srpm` and `rpm` files. Is that correct?

So build system is meson, Makefile is just a facade to meson for users, which are not familiar with meson.

But RPM build is standalone, because meson is execute within the RPM build process, that's why we have those build scripts.

So as Makefile is only a facade, I think it's OK that we are calling RPM build scripts from Makefile

**Comment by engelmi** [comment]
> @engelmi @mwperina Thanks for the comments. I was thinking like: "makefile triggers the real build system (whatever it is)" which in this case is `meson` and generates the `srpm` and `rpm` files. Is that correct?
>

Yes, you are correct. 
 
> So build system is meson, Makefile is just a facade to meson for users, which are not familiar with meson.
> 
> But RPM build is standalone, because meson is execute within the RPM build process, that's why we have those build scripts.
> 
> So as Makefile is only a facade, I think it's OK that we are calling RPM build scripts from Makefile

Yes, the RPM builds are standalone, but its still a bit inconsistent if we call it directly from the Makefile - since it is only a facade for meson, but bypasses it for the RPM build. Not a big issue, I think, so both approaches are fine with me.

**Comment by dougsland** [comment]
> > @engelmi @mwperina Thanks for the comments. I was thinking like: "makefile triggers the real build system (whatever it is)" which in this case is `meson` and generates the `srpm` and `rpm` files. Is that correct?
> 
> Yes, you are correct.
> 
> > So build system is meson, Makefile is just a facade to meson for users, which are not familiar with meson.
> > But RPM build is standalone, because meson is execute within the RPM build process, that's why we have those build scripts.
> > So as Makefile is only a facade, I think it's OK that we are calling RPM build scripts from Makefile
> 
> Yes, the RPM builds are standalone, but its still a bit inconsistent if we call it directly from the Makefile - since it is only a facade for meson, but bypasses it for the RPM build. Not a big issue, I think, so both approaches are fine with me.

Great, so let's move with what we have now which ready to go/merge. :)

File: Makefile [hunk 1]
```diff
@@ -48,5 +48,12 @@ clean:
 	find . -name \*# -delete
 	meson setup --wipe $(BUILDDIR)
 
+rpm: srpm
+	@build-scripts/build-rpm.sh
+
+srpm: 
+	@rpmbuild || (echo "For building RPM package, rpmbuild command is required. To install use: dnf install rpm-build"; exit 1)
+	@build-scripts/build-rpm.sh
```
Comment (pypingou) [inline Makefile]: Does this really build a srpm? It seems to be doing the same thing as `make rpm`

Comment (mwperina) [reply Makefile]: Good catch, I missed that :-(

**Review by mwperina — CHANGES_REQUESTED** [review]
Please check .src.rpm is built properly using `make srpm`

File: Makefile (hunk 1 above)
Comment (dougsland) [reply Makefile]: pingou++ mperina++ me-- ;-)

**Comment by dougsland** [comment]
@mwperina @engelmi @pypingou please re-review. Thanks!

**Comment by rhatdan** [comment]
LGTM

**Review by engelmi — APPROVED** [review]
LGTM

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**